### PR TITLE
chore: increase ivf assignment parallism during indexing

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: rust/lance/src/index/vector/ivf/builder.rs [hunk 1]
```diff
@@ -67,8 +67,9 @@ pub async fn shuffle_dataset(
             // TODO: Make CPU bound to a future.
             ivf.partition_transform(&batch, column).await
         })
-        .buffer_unordered(num_cpus::get())
-        .and_then(|batch| async move {
+        .buffer_unordered(num_cpus::get() * 2)
+        .map(|batch| async move {
+            let batch = batch.unwrap();
```
Comment (westonpace) [inline rust/lance/src/index/vector/ivf/builder.rs:72]: Is `batch` guaranteed to be ok?  Can we use `and_then` here instead?

File: rust/lance/src/index/vector/ivf.rs [hunk 2]
```diff
@@ -779,12 +748,23 @@ pub async fn build_ivf_pq_index(
...
+            ivf_model.centroids.as_ref().try_into()?,
+            metric_type,
+            vec![],
+        );
         // Compute the residual vector to train Product Quantizer.
-        let part_ids = span!(Level::INFO, "compute partition")
-            .in_scope(|| ivf_model.compute_partitions(&training_data, metric_type));
+        let part_ids = ivf2
+            .compute_partitions(training_data.clone())
+            .instrument(span!(Level::INFO, "compute partition for PQ training"))
```
Comment (westonpace) [inline rust/lance/src/index/vector/ivf.rs:762]: Minor nit, this is correct, but could you put the `instrument` call inside of `compute_partitions`?  `instrument` needs to accompany `spawn_blocking` so having those two calls in the same scope makes it easier for future readers to understand why it is there.

File: rust/lance/src/index/vector/ivf/builder.rs [hunk 3]
```diff
@@ -88,8 +89,9 @@ pub async fn shuffle_dataset(
                     })
                     .collect::<Result<Vec<_>>>()
             })
-            .await?
+            .await
         })
+        .buffer_unordered(num_cpus::get())
```
Comment (westonpace) [inline rust/lance/src/index/vector/ivf/builder.rs:94]: I'm not certain but I think you will be ok with a single `buffer_unordered` call at the end of the pipeline.  Having two doesn't hurt though.

**Review by westonpace — APPROVED** [review]
A few minor questions but looks good to me

File: rust/lance/src/index/vector/ivf/builder.rs (hunk 1 above)
Comment (eddyxu) [reply rust/lance/src/index/vector/ivf/builder.rs:72]: Sorry, this was sloppy. Fixed.

File: rust/lance/src/index/vector/ivf.rs (hunk 2 above)
Comment (eddyxu) [reply rust/lance/src/index/vector/ivf.rs:762]: Sure thing.

**Pull request merged**